developers/contracts/xlink-staking.md

```diff
@@ -0,0 +1,458 @@
+
+This setup implies that end users can only access these features indirectly through intermediary contracts (façade, endpoint, etc.) that must be enabled DAO extensions. The external call from the extension to the `xlink-staking` contract, a.k.a. Staking Manager, can be implemented in two ways, resulting in different authentications for the `tx-sender`, the staker.
+
+In summary, these operations are typically accessed by users through intermediary contracts registered as DAO extensions. However, they can also be directly called by a proposal contract executed by the DAO.
+
+#### End user call
+
+```mermaid
+flowchart LR
+    User --Sends transtaction--> i["DAO extension"]
+    i["DAO extension"] -.-> B((Call type))
+    B --contract-call--> sm1[Staking Manager]
+    B --as-contract--> sm2[Staking Manager]
+    sm1 -.-o box1[["**tx-sender**: User
+    **contract-caller**: DAO extension"]]
+    sm2 -.-o box2[["**tx-sender** & **contract-caller**: DAO extension"]]
+```
+
+The `as-contract` call type is the most common approach for liquid staking implementations, where the intermediary contract stakes on behalf of users and provides them with a receipt in the form of a new (rebasing) token.
+
+#### Governance call
+
+```mermaid
+flowchart LR
+    Operator --Sends transtaction--> dao["Executor DAO"]
+    dao --as-contract--> i["Proposal 
+    contract"]
+    i --contract-call--> sm[Staking Manager]
+    sm -.-o box[["**tx-sender**: Executor DAO 
+    **contract-caller**: Proposal contract"]]
+```
+
+### Mechanics
+
+In the Staking Manager, the staker is represented by the `tx-sender`. As explained above, depending on the intermediary contract design, the staker may either be an end user or the endpoint/façade acting on behalf of the user. Therefore, every reference to "staker" (or `user`, as defined within the contract) applies to any of these two possibilities.
+
+Upon staking, the shares corresponding to the amount being staked are calculated and stored in the [`user-shares`](#user-shares) map. This is the staking position and represents the user's portion of the total amount staked. During this operation, the amount to stake is transferred from the `tx-sender` to the Staking Manager, and since the total amount staked and total shares increased, the contract storage is updated to reflect the new state. Refer to the [`stake`](#stake) function for detailed information.
+
+Unstaking performs the inverse operations. It first calculates the shares that correspond to the amount willing to unstake. Then, the unstaked amount is transferred from the Staking Manager to the `tx-sender` and, finally, state upates are performed to decrease user shares, total shares and total amount staked.
+
+{% hint style="info" %}
+Note that both **staking** and **unstaking** operations involve shares management and mutate the staking status of a token in three dimensions: total shares, total amount staked and user's share-based staking position.
```

Comment:
what do you mean by: "mutate the staking status of a token" ? 
which token's status is mutated?

Reply:
Perhaps something worth mention is:
Staking and unstaking shouldn't affect share's price

Reply:
Done 👍

Reply:
> Note that both **staking** and **unstaking** operations involve shares management and update the staking status of a token in three dimensions: total shares, total amount staked and user's share-based staking position.

By "mutate the staking status of a token" I mean that the hole staking state of the involved token is modified: new shares, new staked amount and modifications on staker positions. Changed `mutate` by `update`.

Reply:
s/status of a token in/status of **the** token in/